addons/road-generator/plugin.gd

```diff
@@ -227,6 +236,18 @@ func _handle_gui_select_mode(camera: Camera, event: InputEvent) -> bool:
 		elif _press_init_pos != event.position:
 			# TODO: possibly add min distance before treated as a drag
 			# (does built in godot have a tolerance before counted as a drag?)
+			if _snapping:
+				# Snap selected object to RoadPoint
+				var sel_rp = _nearest_edges[0]
+				var tgt_rp = _nearest_edges[1]
+				selected.snap_to_road_point(sel_rp, tgt_rp)
+				_connect_rp_on_click(tgt_rp, sel_rp)
```

Comment:
Will need to parse this out into a do/undo operation similar to others below. Even if we don't fully implement it yet, could be nice to at least set up the function so we remember to manage it more later. The undo will be a little trickier, since the "do" part is a method, but it should be reasonable to accomplish. I do like having the snap function be on the selected node since that means it'll be accessible to the game on runtime (which is what we'll likely use in wheelsteal for placements/connections)